### ISSUE TITLE
Detect name collisions around _bucket, _count, _sum, and the quantile label during registration

### ISSUE THREAD

**Issue by beorn7** [prometheus/client_golang#426]
With #222 done, we can now detect above mentioned collision during registration time.

See #108 for more details.

**Comment by jeffwidman** [comment]
This implies #222 is done, but confusingly that issue is not marked as close... is this still blocked by work that needs to happen in #222?

**Comment by beorn7** [comment]
Yeah, that's confusing indeed. I don't know what I thought when I wrote that. I'll try to improve the wording.

**Comment by beorn7** [comment]
I mostly assigned this to myself as a “default” because I used to be the maintainer of this repo. Therefore, I'm un-assigning this from myself no. New maintainers @bwplotka & @kakkoyun, please deal with this as you see fit.
